pip/req/req_install.py

```diff
@@ -1014,6 +1024,36 @@ def get_dist(self):
             project_name=dist_name,
             metadata=metadata)
 
+    @property
+    def has_hash_options(self):
+        """Return whether any known-good hashes are specified as options.
+
+        These activate --require-hashes mode; hashes specified as part of a
+        URL do not.
+
+        """
+        return bool(self.options.get('hashes', {}))
+
+    def hashes(self, trust_internet=True):
+        """Return a hash-comparer that considers my option- and URL-based
+        hashes to be known-good.
+
+        Hashes in URLs are almost peers with ones from flags. They satisfy
+        --require-hashes (whether it was implicitly or explicitly activated)
```

Comment:
That's "hashes *in URLs*", meaning ones literally embedded in URLs in requirements files. Hashed from the network are indeed not good enough. I'll clarify the docstring.